### DISCUSSION TITLE
Sourcing blueprint information from the forums

### DISCUSSION THREAD

**Discussion by rianadon** [home-assistant/architecture#601] [General] [answered]
I've been pondering the idea of recommending blueprints in the Home Assistant interface. I've noticed many are created for specific zigbee / zwave devices, and it would be very convenient to, say once you've set up a device, see someone already created a blueprint for it!

My naive solution is to write an integration to use the discourse api to have each instance crawl all the posts. However, something more centralized would put less pressure on the api. Would such a task be apt for being generated on the Home Assistant website (maybe the nabu casa cloud site or I see the [Examples](https://www.home-assistant.io/examples) page in the docs fetches blueprints). Or is something similar already being developed for Blueprints Exchange v2?

**Comment by farmio** [comment]
It would be nice if integrations could bundle some blueprints that are installed with the integration. Maybe have a custom label to identify them. This would ensure some kind of quality control, prevent "spam" and even make them unit-testable and update them with the integration.

**Reply by rianadon** [reply]
I like the idea of quality control and having updates tied to Home Assistant releases & bundled in the repository, but then how would maintenance work?

Would the onus be on integration owners to update blueprints every release? Or would blueprint authors be responsible for creating pull requests to add their blueprints to the core repository?

Compared to an automated system, a manual solution is putting extra work on either integration authors or pull request reviewers.

**Comment by frenck** [comment, marked as answer]
This idea didn't gain enough traction in recent years and is running stale.

I'm going to close it for now; we can reconsider it in the future if this idea becomes relevant again.

../Frenck
